File: HPCore/src/main/kotlin/leakvid/hpcore/domain/Encounter.kt
package leakvid.hpcore.domain

import leakvid.hpcore.domain.enumtypes.DamageType
import leakvid.hpcore.domain.enumtypes.ResistanceType
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document
data class Encounter(
        @Id val name: String,
        val challengeRating: Double,
        val armorClass: Int,
        val HP: Int,
        val strength: Int,
        val dexterity: Int,
        val constitution: Int,
        val intelligence: Int,
        val wisdom: Int,
        val charisma: Int,
        val moveSpeed: Int,
        val resistances: Map<DamageType, ResistanceType>,
        val skills: List<Skill>,
        val features: List<Feature>,
        val actions: List<Action>)

File: app/models/encounter.js
import Model, { attr, hasMany } from '@ember-data/model';

export default class EncounterModel extends Model {
  @attr('string') name;
  @attr('number') challengeRating;
  @attr('number') armorClass;
  @attr('number') hitPoints;
  @attr('number') strength;
  @attr('number') dexterity;
  @attr('number') constitution;
  @attr('number') intelligence;
  @attr('number') wisdom;
  @attr('number') charisma;
  @attr('number') moveSpeed;
  @attr actions;
  @attr features;
  @attr resistances;
  @attr resources;
  @attr skills;
}
